pkg/controller/nodemaintenance/check_lease_supported_test.go

```diff
@@ -0,0 +1,43 @@
+package nodemaintenance
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+    . "github.com/onsi/ginkgo/extensions/table"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+
+var _ = Describe("checkIfLeaseAPISupported", func() {
+
+	DescribeTable("check lease supported",
+		// common test function invoked by table test.
+        func(initWithLeaseAPIPackage bool, leaseSupported bool) {
+				var fcd *FakeClient
+				var cl client.Client
+
+				if initWithLeaseAPIPackage {
+					fcd, cl = createFakeClientsWithLeaseAPIPackage()
+				} else {
+					fcd, cl = createFakeClientsWithoutLeaseAPIPackage()
+				}
+
+				// Create a ReconcileNodeMaintenance object with the scheme and fake client
+				r := &ReconcileNodeMaintenance{client: cl}
+				err := initDrainer(r, fcd)
+
+				// check that initDrainer completed without error.
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(r.isLeaseSupported).To(Equal(leaseSupported))
+
+
+        },
+        Entry("withLeases", true, true),
+        Entry("withouLeases", false, false),
+	)
```

Comment:
I'm not seeing a list of inputs and expected outputs, with a loop to process them.

Reply:
This approach does not scale well.  For example, the test I asked for that covers API server errors.

Reply:
I have added the test for API server errors. Also the mock setup function and error object / success is now passed as argument to the test function, as requested.